test/crossbrowser/src/browser/api.ts

```diff
@@ -0,0 +1,362 @@
+import {
+  ResourceID,
+  add,
+  lookup,
+  remove
+} from './resources';
+
+import {
+  serializeLocalParticipant,
+  serializeLocalTrack,
+  serializeLocalTrackPublication,
+  serializeRoom
+} from './serialize';
+
+declare const Twilio: any;
+
+/**
+ * Connect to a {@link Room}.
+ * @param {Array<*>} args
+ * @param {(room: Room) => void} sendRoomEvents
+ * @returns {Promise<object>}
+ */
+export async function connect(args: any, sendRoomEvents: (room: any) => void): Promise<any> {
+  const [ token, options ] = args;
+  let room: any;
+
+  try {
+    room = await Twilio.Video.connect(token, options);
+  } catch (e) {
+    return {
+      error: {
+        code: e.code,
+        message: e.message
+      }
+    };
+  }
+
+  add(room);
+  add(room.localParticipant);
+  room.localParticipant.tracks.forEach(add);
+  room.localParticipant.trackPublications.forEach(add);
+
+  room.participants.forEach((participant: any) => {
+    add(participant);
+    participant.tracks.forEach(add);
+  });
+  sendRoomEvents(room);
+
+  return {
+    result: serializeRoom(room)
+  };
+}
+
+/**
+ * Create a {@link LocalTrack}.
+ * @param {Array<*>} args
+ * @returns {Promise<object>}
+ */
+export async function createLocalTrack(args: any): Promise<any> {
+  const [ kind, options ] = args;
+  const createLocalTrack: (options: any) => Promise<any> = {
+    audio: (options: any) => Twilio.Video.createLocalAudioTrack(options),
+    data: (options: any) => new Twilio.Video.LocalDataTrack(options),
+    video: (options: any) => Twilio.Video.createLocalVideoTrack(options)
+  }[kind];
+  let localTrack: any;
+
+  try {
+    localTrack = await createLocalTrack(options);
+  } catch (e) {
+    return {
+      error: {
+        message: e.message
+      }
+    };
+  }
+  add(localTrack);
+
+  return {
+    result: serializeLocalTrack(localTrack)
+  };
+}
+
+/**
+ * Create an array of {@link LocalTrack}s.
+ * @param {Array<*>} args
+ * @returns {Promise<object>}
+ */
+export async function createLocalTracks(args: any): Promise<any> {
+  const [ options ] = args;
+  let localTracks: Array<any>;
+
+  try {
+    localTracks = await Twilio.Video.createLocalTracks(options);
+  } catch (e) {
+    return {
+      error: {
+        message: e.message
+      }
+    };
+  }
+  localTracks.forEach(add);
+
+  return {
+    result: localTracks.map(serializeLocalTrack)
+  };
+}
+
+/**
+ * Disconnect from a {@link Room}.
+ * @param {ResourceID} target - Resource ID of the {@link Room}
+ * @returns {object}
+ */
+export function disconnect(target: ResourceID): any {
+  const room: any = lookup(target);
+  if (!room) {
+    return {
+      error: {
+        message: 'Room not found'
+      }
+    };
+  }
+  room.disconnect();
+  const serializedRoom = serializeRoom(room);
+  remove(room);
```

Comment:
Shouldn't there be a Room-specific `remove` that removes the LocalParticipant, RemoteParticipant, etc.